lib/addons/src/transitional-decorator.js

```diff
@@ -0,0 +1,40 @@
+import deprecate from 'util-deprecate';
+
+// Create a decorator that can be used both in the (deprecated) old "hoc" style:
+//   .add('story', decorator(options)(() => <Story />));
+//
+// And in the new, "parameterized" style:
+//   .addDecorator(decorator)
+//   .add('story', () => <Story />, { name: { parameters } });
+
+export const makeTransitionalDecorator = ({ name, parameterName, wrapper }) => {
+  const decorator = options => (getStory, context) => {
+    const parameters = context.parameters && context.parameters[parameterName];
+
+    return wrapper(getStory, context, {
+      options,
+      parameters,
+    });
+  };
+
+  return (...args) => {
+    // Used without options as .addDecorator(decorator)
+    if (typeof args[0] === 'function') {
+      return decorator()(...args);
+    }
+
+    return (...innerArgs) => {
+      // Used as [.]addDecorator(decorator(options))
+      if (typeof innerArgs.length > 1) {
```

Comment:
typeof?

Reply:
wow :0

Reply:
I need to add a test for this modality, it just doesn't really make sense for `notes`. I'll do another PR for `backgrounds` and we'll definitely have one.